src/rdkafka_request.c

```diff
@@ -476,93 +476,177 @@ rd_kafka_FindCoordinatorRequest(rd_kafka_broker_t *rkb,
 }
 
 
-
 /**
- * @brief Parses a ListOffsets reply.
- *
- * Returns the parsed offsets (and errors) in \p offsets which must have been
- * initialized by caller.
- *
- * @returns 0 on success, else an error (\p offsets may be completely or
- *          partially updated, depending on the nature of the error, and per
- *          partition error codes should be checked by the caller).
- */
-static rd_kafka_resp_err_t
-rd_kafka_parse_ListOffsets(rd_kafka_buf_t *rkbuf,
-                           rd_kafka_topic_partition_list_t *offsets) {
+ * @brief Parses the response given the rd_kafka_buf_t and returns the rd_list_t with the element
+ *        as rd_kafka_ListOffsetsResultInfo_t which encapsulates all the Info about a topic_partition offset for 
+ *        which ListOffset request was made.
+ * @param reply The rd_kafka_buf_t to be parsed for the response of ListOffsets.
+ * @param errstr The errstr to be populated if the parser fails.
+ * @param errstr_size The size of errstr.
+*/
+rd_list_t *rd_kafka_ListOffsetsResponseParser(rd_kafka_buf_t *reply,char *errstr,size_t errstr_size){
```

Comment:
use previous `rd_kafka_parse_ListOffsets` instead, upgrading it.

Reply:
I find this one much cleaner and better.